packages/crab-netting/src/CrabNetting.sol

```diff
@@ -7,49 +9,126 @@ contract CrabNetting {
-        IERC20(usdc).transferFrom(msg.sender, address(this), amount);
-        usd_balance[msg.sender] = usd_balance[msg.sender] + amount;
-        deposits.push(Receipt(msg.sender, amount));
+    function depositUSDC(uint256 _amount) public {
+        IERC20(usdc).transferFrom(msg.sender, address(this), _amount);
+        usd_balance[msg.sender] = usd_balance[msg.sender] + _amount;
+        deposits.push(Receipt(msg.sender, _amount));
     }
 
-    function withdrawUSDC(uint256 amount) public {
-        require(usd_balance[msg.sender] >= amount);
-        usd_balance[msg.sender] = usd_balance[msg.sender] - amount;
-        IERC20(usdc).transfer(msg.sender, amount);
+    function withdrawUSDC(uint256 _amount) public {
+        // TODO ensure final version does not need this check
+        //require(usd_balance[msg.sender] >= _amount, "Withdrawing more than balance");
+        usd_balance[msg.sender] = usd_balance[msg.sender] - _amount;
+        IERC20(usdc).transfer(msg.sender, _amount);
     }
 
-    function depositCrab(uint256 amount) public {
-        IERC20(crab).transferFrom(msg.sender, address(this), amount);
-        crab_balance[msg.sender] = crab_balance[msg.sender] + amount;
-        crab_deposits.push(Receipt(msg.sender, amount));
+    function depositCrab(uint256 _amount) public {
+        IERC20(crab).transferFrom(msg.sender, address(this), _amount);
+        crab_balance[msg.sender] = crab_balance[msg.sender] + _amount;
+        withdraws.push(Receipt(msg.sender, _amount));
     }
 
-    function withdrawCrab(uint256 amount) public {
-        require(crab_balance[msg.sender] >= amount);
-        crab_balance[msg.sender] = crab_balance[msg.sender] - amount;
-        IERC20(crab).transfer(msg.sender, amount);
+    function withdrawCrab(uint256 _amount) public {
+        // TODO ensure final version does not need this check
+        // require(crab_balance[msg.sender] >= _amount);
+        crab_balance[msg.sender] = crab_balance[msg.sender] - _amount;
+        IERC20(crab).transfer(msg.sender, _amount);
     }
 
-    function balanceOf(address account) public view returns (uint256) {
-        return usd_balance[account];
+    function netAtPrice(uint256 _price, uint256 _quantity) public {
+        // TODO make only owner
+        uint256 forWithdraw = _quantity;
+        require(
+            _quantity <= IERC20(usdc).balanceOf(address(this)),
+            "Not enough deposits to net"
+        );
+        // TODO DO we need to do the same check for crab withdrawals?
+        // can we just do deposits alone and forget about withdrawals netting?
+
+        // TODO make sure we dont need SafeMath in final version
+        uint256 i = depositsIndex;
+        while (_quantity > 0 && i < deposits.length) {
+            Receipt memory deposit = deposits[i];
+            if (deposit.amount < _quantity) {
+                _quantity = _quantity - deposit.amount;
+                usd_balance[deposit.sender] =
```

Comment:
Consider this situation
- I deposit 100 USDC, now the receipt will have 100 as amount and usdc_balance will be 100 
- I withdraw 50 USDC, now the receipt will still be 100 but usdc_balance will be 50
- this operation will lead to 50 - 100 and will throw error